main.go

```diff
@@ -132,3 +146,54 @@ func printThemes(cmd *cobra.Command) {
 		cmd.Println()
 	}
 }
+
+type numberWriter struct {
+	w           io.Writer
+	currentLine int64
+	buf         []byte
+}
+
+func (w *numberWriter) Write(p []byte) (n int, err error) {
+
+	var (
+		original = p
+		tokenLen int
+	)
+	for i, c := range original {
```

Comment:
[ask]Is it better to check if it continats `\n` before loop ❓ 

```suggestion
if bytes.Contains(p, []byte(string('\n'))) {
	for i, c := range original {
	   ...
```

Reply:
I'll use "strings.Split" to rewrite it a bit more elegantly

Reply:
I realized that I couldn't fix it that elegantly by using "bytes.Split".
I've added early returns for clarity, but note that adding "bytes.Contains" has increased the amount of slice `p` scanning.
Original: The scan amount is always O(N)
New implementation: scan volume is at worst O(2N) (if the data does not contain line breaks)